frontend/src/pages/add-blog.tsx

```diff
@@ -198,11 +195,10 @@ function AddBlog() {
             {CATEGORIES.map((category) => (
               <span
                 key={category}
-                className={`cursor-pointer
-									${
-                    formData.categories.includes(category)
-                      ? categoryProps(category, true)
-                      : categoryProps(category, false)
+                className={`${isDisable && !formData.categories.includes(category) ? "cursor-not-allowed" : "cursor-pointer"}
+									${formData.categories.includes(category)
+                    ? categoryProps(category, true, isDisable ? true : false)
```

Comment:
Can you kindly optimize this even further, refactoring it to a new function which would simplify this process further.

Reply:
Ok @krishnaacharyaa  I will do that